src/database/common.c

```diff
@@ -19,7 +19,10 @@
 #include "files.h"
 
 sqlite3 *FTL_db;
-bool database = false;
+// This boolean is set to false once we hit the
+// first database error to prevent further access
+// to the pihole-FTL.db database
+bool database = true;
```

Comment:
This check in main should now be moved into `db_init` as the `database` value will remain `true` if the check fails even though the database is not available.
https://github.com/pi-hole/FTL/blob/9f127e9838c172d5557b969c56164f0627953e86/src/main.c#L70-L72